File: 1/1.rb
def part1
  input = parse_input

  sum = input[0..-2].each_with_index.inject(0) do |sum, (n, i)|
    input[i+1] == n ? sum += n : sum
  end

  sum += input[-1] if input[-1] == input[0]

  sum.to_i
end

def part2
  input = parse_input
  flipped_input = input.rotate(input.length/2)

  sum = input.each_with_index.inject(0) do |sum, (n, i)|
    flipped_input[i] == n ? sum += n : sum
  end

  sum.to_i
end

def parse_input
  ARGV[0].split("").map(&:to_i)
end

puts "Part 1: #{part1()}"
puts "Part 2: #{part2()}"

File: 1/1.exs
defmodule Day1 do
  def part1 do
    input = parse_input()
    sum = input
    |> Enum.slice(0..-2)
    |> Enum.with_index
    |> Enum.reduce(0, fn ({n, i}, sum) ->
      if Enum.at(input, i+1) == n, do: sum + n, else: sum
    end)

    last = Enum.at(input, -1)

    if(last == hd(input), do: sum + last, else: sum)
    |> IO.puts
  end

  defp parse_input do
    [input] = System.argv

    input
    |> String.split("", trim: true)
    |> Enum.map(&String.to_integer/1)
  end
end

Day1.part1
